core/kogitoservice/deployer_resources.go

```diff
@@ -239,38 +244,81 @@ func (s *serviceDeployer) getComparator() compare.MapComparator {
 	return compare.MapComparator{Comparator: resourceComparator}
 }
 
-func (s *serviceDeployer) fetchKogitoInfraProperties() (map[string]string, []corev1.EnvVar, []v1beta1.KogitoInfraVolume, error) {
+func (s *serviceDeployer) fetchKogitoInfraProperties() (map[string]string, []corev1.EnvVar, []api.KogitoInfraVolumeInterface, error) {
 	kogitoInfraReferences := s.instance.GetSpec().GetInfra()
-	log.Debug("Going to fetch kogito infra properties", "infra", kogitoInfraReferences)
+	s.Log.Debug("Going to fetch kogito infra properties", "infra", kogitoInfraReferences)
 	consolidateAppProperties := map[string]string{}
 	var consolidateEnvProperties []corev1.EnvVar
-	var volumes []v1beta1.KogitoInfraVolume
+	var volumes []api.KogitoInfraVolumeInterface
 	for _, kogitoInfraName := range kogitoInfraReferences {
 		// load infra resource
-		kogitoInfraInstance, err := infrastructure.MustFetchKogitoInfraInstance(s.client, kogitoInfraName, s.instance.GetNamespace())
+		infraManager := manager.NewKogitoInfraManager(s.Context, s.infraHandler)
+		kogitoInfraInstance, err := infraManager.MustFetchKogitoInfraInstance(types.NamespacedName{Name: kogitoInfraName, Namespace: s.instance.GetNamespace()})
 		if err != nil {
 			return nil, nil, nil, err
 		}
 
 		runtime := s.instance.GetSpec().GetRuntime()
 
 		// fetch app properties from Kogito infra instance
-		appProp := kogitoInfraInstance.Status.RuntimeProperties[runtime].AppProps
+		appProp := kogitoInfraInstance.GetStatus().GetRuntimeProperties()[runtime].GetAppProps()
```

Comment:
Getting nil pointer error when on this line when deploying KogitoRuntime with KogitoInfra referencing Knative Broker:
```
E0222 14:32:53.431166       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 600 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x18b5d40, 0x2bd9730)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/runtime/runtime.go:48 +0x82
panic(0x18b5d40, 0x2bd9730)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/kiegroup/kogito-cloud-operator/core/kogitoservice.(*serviceDeployer).fetchKogitoInfraProperties(0xc0007dc0e0, 0x1af6b2c, 0x1d, 0x0, 0x0, 0x0, 0x22, 0x1adbcc4, 0x6, 0xc000c2be20)
	/workspace/core/kogitoservice/deployer_resources.go:264 +0x3b4
github.com/kiegroup/kogito-cloud-operator/core/kogitoservice.(*serviceDeployer).createRequiredResources(0xc0007dc0e0, 0x0, 0x0, 0x1)
```

Reply:
reported as https://issues.redhat.com/browse/KOGITO-4529